src/main/java/org/spongepowered/common/data/manipulator/immutable/item/ImmutableSpongeDurabilityData.java

```diff
@@ -0,0 +1,98 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.manipulator.immutable.item;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.spongepowered.common.data.util.ComparatorUtil.intComparator;
+
+import com.google.common.collect.ComparisonChain;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.MemoryDataContainer;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.item.ImmutableDurabilityData;
+import org.spongepowered.api.data.manipulator.mutable.item.DurabilityData;
+import org.spongepowered.api.data.value.immutable.ImmutableBoundedValue;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.common.data.manipulator.immutable.common.AbstractImmutableData;
+import org.spongepowered.common.data.manipulator.mutable.item.SpongeDurabilityData;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeBoundedValue;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+
+public class ImmutableSpongeDurabilityData extends AbstractImmutableData<ImmutableDurabilityData, DurabilityData> implements ImmutableDurabilityData {
+
+    final int durability;
+    final ImmutableValue<Boolean> unbreakable;
+
+    public ImmutableSpongeDurabilityData(int durability, boolean unbreakable) {
+        super(ImmutableDurabilityData.class);
+        checkArgument(durability >= 0);
+        this.durability = durability;
+        this.unbreakable = ImmutableSpongeValue.cachedOf(Keys.UNBREAKABLE, false, unbreakable);
+        this.registerGetters();
+    }
+
+    public ImmutableSpongeDurabilityData() {
+        this(60, false);
+    }
+
+    @Override
+    protected void registerGetters() {
+        registerFieldGetter(Keys.ITEM_DURABILITY, () -> this.durability);
+        registerKeyValue(Keys.ITEM_DURABILITY, this::durability);
+
+        registerFieldGetter(Keys.UNBREAKABLE, this.unbreakable::get);
+        registerKeyValue(Keys.UNBREAKABLE, this::unbreakable);
+    }
+
+    @Override
+    public ImmutableBoundedValue<Integer> durability() {
+        return new ImmutableSpongeBoundedValue<Integer>(Keys.ITEM_DURABILITY, this.durability, 60, intComparator(), 0, Integer.MAX_VALUE);
```

Comment:
Use a final class variable so you don't create this object each time.

Also, use the `SpongeValueBuilder` as well.